### PR TITLE
Enable private repositories

### LINKED ISSUE #35: Support imagepullsecrets from Pod definition
We need to support imagepullsecrets for private repo access

### PR DESCRIPTION
**Pull request by lachie83**
Closes #35

### CODE REVIEW

File: synchronizer.ts [hunk 1]
```diff
@@ -27,6 +27,38 @@ export async function Synchronize(client: api.Core_v1Api, startTime: Date, rsrcC
             }
             let containers = new Array<Object>();
             let cPorts = new Array<Object>();
+
+            let imageRegistryCredentials = new Array<Object>();
+            
+            if (pod.spec.imagePullSecrets != null) {
+                let secrets = await client.listNamespacedSecret('default');
```
Comment (brendandburns) [inline synchronizer.ts]: use pod namespace instead of 'default'?

File: synchronizer.ts [hunk 2]
```diff
@@ -27,6 +27,38 @@ export async function Synchronize(client: api.Core_v1Api, startTime: Date, rsrcC
...
+                for (let secret of secrets.body.items) {
```
Comment (brendandburns) [inline synchronizer.ts]: why are you list and searching instead of doing a straight instance get?

File: synchronizer.ts [hunk 3]
```diff
@@ -27,6 +27,38 @@ export async function Synchronize(client: api.Core_v1Api, startTime: Date, rsrcC
...
+                    if (secret.metadata.name == pod.spec.imagePullSecrets.name && secret.metadata.type == "kubernetes.io/dockercfg") {
+                        /*
```
Comment (brendandburns) [inline synchronizer.ts]: delete this

File: synchronizer.ts [hunk 4]
```diff
@@ -94,7 +126,8 @@ export async function Synchronize(client: api.Core_v1Api, startTime: Date, rsrcC
                 tags: {
                     "orchestrator": "kubernetes"
                 },
-                location: region
+                location: region,
+                imageRegistryCredentials: imageRegistryCredentials,
```
Comment (brendandburns) [inline synchronizer.ts]: delete trailing comma.  *sigh* javascript...

**Comment by brendandburns** [comment]
more or less looks right.

thanks.

**Comment by lachie83** [comment]
@brendandburns Can you please review again at your earliest convenience?

**Comment by brendandburns** [comment]
Looks great. Thanks! Merging.

**Pull request merged**